### PR TITLE
NFDIV-943: Update contact details

### PR DESCRIPTION
**Pull request by SomeshRA**
### JIRA link (if applicable) ###
NFDIV-943: Update contact details

### Change description ###
Update contact details

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateContactDetails.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            .explicitGrants()
+            .grant(CREATE_READ_UPDATE,
+                CASEWORKER_COURTADMIN_CTSC,
+                CASEWORKER_COURTADMIN_RDU)
+            .grant(READ,
+                SOLICITOR,
+                CASEWORKER_SUPERUSER,
+                CASEWORKER_LEGAL_ADVISOR));
+
+        pageDetails(pageBuilder);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateContactDetails.java:47]: page fields should go in a separate class if you check other events

Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerUpdateContactDetails.java]: Created a new page class

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/UpdateContactDetails.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        FieldCollection.FieldCollectionBuilder<CaseData, State, Event.EventBuilder<CaseData, UserRole, State>> respondentFieldBuilder
+            = getRespFieldBuilder(applicantFieldBuilder);
+
+        respondentFieldBuilder.label("Label-UpdateMarriageCertWarning",
+            "### Only update Marriage Certificate Names to make them match the marriage certificate exactly")
+            .complex(CaseData::getApplication)
+            .complex(Application::getMarriageDetails)
+            .optional(MarriageDetails::getApplicant1Name)
+            .optional(MarriageDetails::getApplicant2Name)
+            .done()
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/UpdateContactDetails.java:34]: I think you lost the formatting when you moved this out

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/UpdateContactDetails.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+import uk.gov.hmcts.divorce.common.model.State;
+import uk.gov.hmcts.divorce.common.model.UserRole;
+
+public class UpdateContactDetails  implements CcdPageConfiguration {
+    @Override
+    public void addTo(PageBuilder pageBuilder) {
+        FieldCollection.FieldCollectionBuilder<CaseData, State, Event.EventBuilder<CaseData, UserRole, State>> fieldCollectionBuilder
+            = pageBuilder.page("UpdateContactDetails");
+
+        FieldCollection.FieldCollectionBuilder<CaseData, State, Event.EventBuilder<CaseData, UserRole, State>> applicantFieldBuilder
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/UpdateContactDetails.java:23]: you don't need to return these as new properties - you can just use `fieldCollectionBuilder` the whole way through. I would rename `getApplicantFieldBuilder` and `getRespFieldBuilder` to `addApplicant1Fields` and `addRespondentFields`

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/UpdateContactDetails.java (hunk 2 above)
Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/UpdateContactDetails.java]: Formatted the code

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/UpdateContactDetails.java (hunk 3 above)
Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/page/UpdateContactDetails.java]: Implemented the suggested changes.

**Review by linusnorton — APPROVED** [review]

**Pull request merged**